app/src/components/adminConsole/currentCup/CreateCupButton.tsx

```diff
@@ -1,6 +1,6 @@
 import React, { useState, useEffect } from "react";
 import { createNewCup, GetCupNames } from "../../../firebase/cups/CupService";
-import Cup from "../../../types/Cup";
+import Cup, { CupWithId } from "../../../types/Cup";
```

Comment:
```suggestion
import Cup from "../../../types/Cup";
```

This should have been replaced with `Entity<Cup>`.